libcudacxx/include/cuda/std/__ranges/iota_view.h

```diff
@@ -157,19 +157,19 @@ class iota_view : public view_interface<iota_view<_Start, _BoundSentinel>>
   _CCCL_TEMPLATE(class _BoundSentinel2 = _BoundSentinel)
   _CCCL_REQUIRES(same_as<_Start, _BoundSentinel2>)
   _CCCL_API constexpr iota_view(__iterator __first, __iterator __last)
-      : iota_view(::cuda::std::move(__first.__value_), ::cuda::std::move(__last.__value_))
+      : iota_view(::cuda::std::move(*__first), ::cuda::std::move(*__last))
   {}
 
   _CCCL_TEMPLATE(class _BoundSentinel2 = _BoundSentinel)
   _CCCL_REQUIRES(same_as<_BoundSentinel2, unreachable_sentinel_t>)
   _CCCL_API constexpr iota_view(__iterator __first, _BoundSentinel __last)
-      : iota_view(::cuda::std::move(__first.__value_), ::cuda::std::move(__last))
+      : iota_view(::cuda::std::move(*__first), ::cuda::std::move(__last))
   {}
 
   _CCCL_TEMPLATE(class _BoundSentinel2 = _BoundSentinel)
   _CCCL_REQUIRES((!same_as<_Start, _BoundSentinel2>) _CCCL_AND(!same_as<_Start, unreachable_sentinel_t>))
   _CCCL_API constexpr iota_view(__iterator __first, __sentinel __last)
-      : iota_view(::cuda::std::move(__first.__value_), ::cuda::std::move(__last.__bound_sentinel_))
+      : iota_view(::cuda::std::move(*__first), ::cuda::std::move(__last.__bound_sentinel_))
```

Comment:
Note for reviewers, I turned `cuda::counting_iterator` into a class, so we cannot access the private member anymore here. I did not want to fight with forward declarations, so just using `operator*` seems reasonable